[jasperng] iP

jasperng-nus opened the pull request:
# Luke
> “Learn from yesterday, live for today, hope for tomorrow. The important thing is not to stop questioning.” – Albert Einstein ([source](https://www.brainyquote.com/quotes/albert_einstein_125368))
- text based
- easy to use
- ~~SUPER~~ *FAST* to use

1. download from here.
2. double click it.
3. add your tasks.
4. let it manage yours tasks for you :+1:

And it is **FREE**!

Features:
- [x] managing task
- [x] add task
- [ ] delete task


> [!NOTE]
> There some things to note.

this is an inline `code` it has back-ticks `around` it

```java
public class Main {
    public static void main(String[] args) {
        System.out.println("Hello World!");
    }
}

```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

jasperng-nus committed:
Level 0. Rename, Greet, Exit

jasperng-nus committed:
Level 1. Echo

jasperng-nus committed:
Level 2. Add, List

jasperng-nus committed:
Level 3. Mark as Done

jasperng-nus committed:
Level 4. ToDos, Events, Deadlines

jasperng-nus committed:
A-TextUiTesting: Automated Text UI Testing

jasperng-nus committed:
Level 5. Handle Errors

jasperng-nus committed:
Level 6. Delete

Austintjh19 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,13 @@
+public class Deadline extends Task {
+    protected String by;
```
Is using protected the most suitable option ? It makes it accessible to other classes from the same package ... I noticed the same issue in several other places too.

Austintjh19 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,16 @@
+public class Event extends Task{
```
Whitespace should proceed { , according to our coding standard. Just something to take note.

Austintjh19 commented on src/main/java/Luke.java:
```diff
@@ -0,0 +1,205 @@
+
+        // Conditions
+        while (!input.equals("bye")) {
+            if (input.equals("list")) {
```
Maybe consider creating functions to execute the commands ? E.g. when input.equals("list ") it call a separate function like outputTasks(). You can possibly perform you validation checks in these functions as well.

Austintjh19 replied:
Consider switching to a case statement it can potentially enhance code readability.

Austintjh19 commented on src/main/java/Luke.java:
```diff
@@ -0,0 +1,205 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Luke {
```
Maybe enhance the class structure by decomposing the functionality into separate classes such as InputValidator for validating user inputs, Outputs for managing output operations, and InputInterpreter for processing user commands, thereby promoting a clearer separation of concerns.

Austintjh19 commented on src/main/java/Luke.java:
```diff
@@ -0,0 +1,205 @@
+                        int index = Integer.parseInt(instruction[1]) - 1;  // array is 0-indexed
+
+                        if (index >= list.size()) {
+                            throw new LukeException("Hold up!! There is no such task in the list.\n"
```
Nice to see you are already incorporating Exceptions.

Austintjh19 commented on src/main/java/TaskException.java:
```diff
@@ -0,0 +1,5 @@
+public class TaskException extends Exception {
```
Maybe have this extend LukeException ? e.g. LukeTaskException. Won't have to check for LukeException | TaskException for each relevant exception.

Austintjh19 commented on src/main/java/Luke.java:
```diff
@@ -0,0 +1,205 @@
+                }
+                System.out.println("________________________________________________________________________");
+
+            } else if (input.contains("mark")) {
```
Might be a redundant if check. You can just split the string and verify if the mark/ unmark is present, while also ensuring the length of the split instruction is 2. This way, you confirm the task name's presence without extra checks. This might be applicable for other commands as well.

Austintjh19 commented on src/main/java/Luke.java:
```diff
@@ -0,0 +1,205 @@
+        // User inputs
+        Scanner sc = new Scanner(System.in);
+        String input = sc.nextLine();
+        ArrayList<Task> list = new ArrayList<>();
```
Possibly a more appropriate variable name to make the code more readable.

Austintjh19 commented on src/main/java/Luke.java:
```diff
@@ -0,0 +1,205 @@
+
+            } else {
+                try {
+                    if (input.equals("todo") || input.equals("deadline") || input.equals("event")) {
```
Maybe consider the case where user enters "todo " or "deadline " or "event ".

Austintjh19 reviewed (commented):
Good start. The basics are there just some modifications to the Luke class are needed.

Austintjh19 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,13 @@
+public class Deadline extends Task {
```
JavaDoc comments of classes and functions would be nice.

Austintjh19 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,16 @@
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + " (from: " + this.from + " to: " + this.to + ")";
+
```
Trailing whitespaces are quite prevalent in your code. Just remove them and you're good.

Austintjh19 reviewed (commented):
Great work on coding standards. Seems like it was well followed.

jasperng-nus committed:
Level 7. Save

jasperng-nus committed:
Merge branch 'branch-Level-7'

jasperng-nus committed:
Level 8. Dates and Times

jasperng-nus committed:
Merge branch 'branch-Level-8'

jasperng-nus committed:
A-MoreOOP: Use More OOP

jasperng-nus committed:
Revert "A-MoreOOP: Use More OOP"

This reverts commit 10347a17d93531ed5bd67579173f055220753aa2.

jasperng-nus committed:
Revert "Revert "A-MoreOOP: Use More OOP""

This reverts commit 5857864f5952807ac6785b34b5ff6b66a164ed50.

jasperng-nus committed:
A-Packages: Organize into Packages

jasperng-nus committed:
Merge branch 'branch-A-Packages'

jasperng-nus committed:
Merge branch 'add-gradle-support'

jasperng-nus committed:
A-Gradle: Use Gradle

jasperng-nus committed:
Merge branch 'add-gradle-support'

jasperng-nus committed:
A-JUnit: Add JUnit Tests

jasperng-nus committed:
Merge branch 'branch-A-JUnit'

jasperng-nus committed:
A-JavaDoc: JavaDoc

jasperng-nus committed:
A-CodingStandard: Follow the Coding Standard

jasperng-nus committed:
Level 9. Find

jasperng-nus committed:
Merge branch 'branch-Level-9'

jasperng-nus committed:
Merge Conflict: combined both the branch-A-JavaDoc and branch-A-CodingStandards with the master branch

ChillinRage commented on src/main/java/luke/parser/Parser.java:
```diff
@@ -0,0 +1,215 @@
+import luke.exception.FileException;
+import luke.exception.LukeException;
+import luke.exception.TaskException;
+import luke.task.*;
```
Would it be better if you split the wildcard import and instead import each module explicitly?

ChillinRage commented on src/main/java/luke/parser/Parser.java:
```diff
@@ -0,0 +1,215 @@
+     * @param ui The user interface.
+     * @param storage The storage of the tasks.
+     */
+    public void parse(TaskList list, Ui ui, Storage storage) {
```
Will a switch-case work better instead of if-else blocks?

ChillinRage commented on src/main/java/luke/parser/Parser.java:
```diff
@@ -0,0 +1,215 @@
+
+                    if (!delete.equals("delete")) {
+                        throw new LukeException("Hold up!! I am sorry, but I don't know what you mean by that :'(");
+                    }
```
How about using ".startsWith("delete")" instead to check for commands? That way, you wouldn't need a separate if equals block.

ChillinRage commented on src/main/java/luke/parser/Parser.java:
```diff
@@ -0,0 +1,215 @@
+                        // +6 to remove the "/from " and start from the index after "/from "
+                        // -1 to remove the space before "/to"
+                        String from = input.substring(input.indexOf("/from") + 6, input.indexOf("/to") - 1);
+                        // +4 to remove the "/to " and start from the index after "/to "
```
I like that you use comments to explain the purpose of your code.

ChillinRage commented on src/main/java/luke/task/Event.java:
```diff
@@ -0,0 +1,59 @@
+package luke.task;
+
+import luke.exception.DateException;
+import luke.parser.DateTimeParser;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
```
I like that you separated the imports by packages.

ChillinRage commented on src/main/java/luke/ui/Ui.java:
Would it be better to have a separate function of constant variable to print the horizontal lines instead?

jasperng-nus committed:
Level 10. GUI

jasperng-nus committed:
Merge conflict: some changes have not been committed.

jasperng-nus committed:
Merge branch 'branch-Level-10'

jasperng-nus committed:
Add checkstyle.xml and supression.xml

Ensuring consistent coding standards is crucial for code quality and readability.

Applied CheckStyle rules to the entire codebase. Resolved CheckStyle violations related to indentation, naming conventions, and coding style.

Enforcing CheckStyle rules ensures a clean and consistent codebase, making it easier to maintain and understand.

jasperng-nus committed:
Merge branch 'A-CheckStyle'

jasperng-nus committed:
Add a handleMarkOrUnmark() method

The readTask() method is too long.

To enhance code maintainability and readability, it is necessary to address the existing method's length and complexity.

Using a new method handleMarkOrUnmark(), to handle the conditional statement to shorten the method length and make it more readable.

jasperng-nus committed:
Add assertions into Storage and Luke classes

The current situation lacks assumption in the Storage and Luke classes.

Introduce assertions at various points to ensure that important assumptions are made before running the rest of the code.

jasperng-nus committed:
Merge pull request #2 from jasperng-nus/branch-A-Assertions

Add assertions into Storage and Luke classes

jasperng-nus committed:
fix: files are the same but shown as different

jasperng-nus committed:
Merge branch 'master' of https://github.com/jasperng-nus/ip

jasperng-nus committed:
Add more methods to separate and shorten the parse method

jasperng-nus committed:
Merge pull request #3 from jasperng-nus/branch-A-CodeQuality

Add more methods to separate and shorten the parse method

jasperng-nus committed:
Add a help page to guide the user on the commands to use

jasperng-nus committed:
Add the Product Screenshot

jasperng-nus committed:
Merge branch 'master' of https://github.com/jasperng-nus/ip

jasperng-nus committed:
Add a User Guide
